### PR TITLE
Split bundle params

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR splits `BundleParams` in order to be able to use `config.Component` independently of `log.Component`. See https://github.com/DataDog/datadog-agent/pull/14538 for an example of why it is important.
This PR also makes  fields for `config.Params` and `log.Params` unexported to enforce the usage of constructors (To easily add mandatory fields).

If you are going to review the whole PR, this is strongly recommended to review commit by commit.

### Describe how to test/QA your changes

This PR should be mostly covered by unit tests but to QA this dev:
- Agent shared component team : Check few commands work as expected.
- Security agent: Check few commands work as expected.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/core/config/params.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		defaultConfPath: defaultConfPath,
+	}
+	for _, o := range options {
+		o(&params)
+	}
+	return params
+}
+
+// NewAgentParams creates a new instance of Params for the Agent.
+func NewAgentParams(confFilePath string, configLoadSecrets bool, options ...func(*Params)) Params {
```
Comment (hush-hush) [inline comp/core/config/params.go:64]: WDYT about having 2 methods for the agent `NewAgentParamsWithSecrets` and `NewAgentParamsWithoutSecret`. We're going to use this in a lot of places and a bool parameter is never very obvious (ex `NewAgentParams("", true)`). Or `NewAgentParams` and `NewAgentParamsWithSecrets` ?

File: comp/core/log/params.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+// LogFileFn returns the log file
+func (params Params) LogFileFn(c configGetter) string {
+	return params.logFileFn(c)
+}
+
+// LoggerName is the name that appears in the logfile
+func (params Params) LoggerName() string {
+	return params.loggerName
+}
```
Comment (hush-hush) [inline comp/core/log/params.go:149]: I think we should align those with `config` and also use the `With` prefix. WDYT ?

**Review by hush-hush — APPROVED** [review]
This is great :+1:  ! I added a few comments/questions. Some could be done in future PR so don't hesitate to ignore and create card for them.

File: comp/core/config/params.go [hunk 3]
```diff
@@ -60,14 +60,22 @@ func NewParams(defaultConfPath string, options ...func(*Params)) Params {
...
+// NewAgentParams creates a new instance of Params for the Agent.
 func NewAgentParams(confFilePath string, configLoadSecrets bool, options ...func(*Params)) Params {
 	params := NewParams(common.DefaultConfPath, options...)
 	params.ConfFilePath = confFilePath
 	params.ConfigLoadSecrets = configLoadSecrets
 	return params
 }
 
+// NewSecurityAgentParams creates a new instance of Params for the Security Agent.
+func NewSecurityAgentParams(securityAgentConfigFilePaths []string, configLoadSecurityAgent bool, options ...func(*Params)) Params {
```
[the remark this answers is not in the document]
Comment (paulcacheux) [reply comp/core/config/params.go:72]: Yes, I don't really see a case where you would provide secagent config paths but you would not want to load them

**Review by paulcacheux — APPROVED** [review]

**Review by davidor — APPROVED** [review]
:+1: for files owned by @DataDog/container-integrations

**Review by BaptisteFoy — APPROVED** [review]
Looks good for Remote Config

**Review by mbotarro — APPROVED** [review]
👍 for files owned by @DataDog/processes

**Review by FlorentClarret — APPROVED** [review]

**Review by ethan-lowman-dd — APPROVED** [review]

**Review by carlosroman — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]
👍 for NDM owned files

File: comp/core/log/params.go (hunk 2 above)
Comment (ogaca-dd) [reply comp/core/log/params.go:149]: The last 3 methods are getters and there is no reason to use `With` prefix for them.

`LogForOneShot` and `LogForDaemon` are constructors for `log.Params` so no reason to prefix with `With`.

`LogToFile` can be a valid candidate for `With` but I am not sure it worths to use the option pattern for a single field (and `LogToFile` is used once in the code). What do you think?

File: comp/core/config/params.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply comp/core/config/params.go]: Code updated.

File: comp/core/config/params.go (hunk 1 above)
Comment (ogaca-dd) [reply comp/core/config/params.go]: Code updated.

**Review by paulcacheux — APPROVED** [review]

File: comp/core/log/params.go (hunk 2 above)
Comment (hush-hush) [reply comp/core/log/params.go:149]: Sorry, I miss that the last 3 were getters. I think it's worth it for `LogToFile`, to at least highlight that it's not a getter like the other ones.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**